api/src/repositories/submission-job-queue-repository.ts

```diff
@@ -0,0 +1,96 @@
+import SQL from 'sql-template-strings';
+import { ApiExecuteSQLError } from '../errors/api-error';
+import { IProprietaryInformation } from '../services/submission-job-queue-service';
+import { BaseRepository } from './base-repository';
+
+export interface ISubmissionJobQueueModel {
+  submission_job_queue_id: number;
+  submission_id: number;
+  job_start_timestamp: string | null;
+  job_end_timestamp: string | null;
+  security_request: string | null; // stored as JSON object
+  create_date: string;
+  create_user: number;
+  update_date: string | null;
+  update_user: number | null;
+  // no attempt count?
+}
+
+export interface IInsertSubmissionJobQueueRecord {
+  submission_job_queue_id: number;
+  submission_id: number;
+}
+
+export class SubmissionJobQueueRepository extends BaseRepository {
+  /**
+   * Creates Job queue and returns the record Id
+   *
+   * @param {number} queueId
+   * @param {number} submissionId
+   * @param {IProprietaryInformation} proprietaryInformation
+   * @return {*}  {Promise<{ queue_id: number }>}
+   * @memberof SubmissionJobQueueRepository
+   */
+  async insertJobQueueRecord(
+    queueId: number,
+    submissionId: number,
+    proprietaryInformation?: IProprietaryInformation
+  ): Promise<{ queue_id: number }> {
+    const sqlStatement = SQL`
+      INSERT INTO submission_job_queue (
+        submission_job_queue_id,
+        submission_id,
+        security_request
+      ) VALUES (
+        ${queueId},
+        ${submissionId},
+        ${JSON.stringify(proprietaryInformation ? proprietaryInformation : {})}
+      )
+      RETURNING submission_job_queue_id;
+    `;
+
+    const response = await this.connection.sql<{ submission_job_queue_id: number }>(sqlStatement);
+    return { queue_id: response.rows[0].submission_job_queue_id };
```

Comment:
Can we check for a row count of 0 and throw an error accordingly?

Reply:
done